content/design/vendoring.md

```diff
@@ -53,3 +53,224 @@ already written):
 You now have a `vendor` directory that contains the sources for `goagen`, compile them and use the
 tool as described above. Don't forget to recompile the tool after each `glide update` that updates
 `goa`.
+
+# Code-based generation and vendoring
+
+> For the impatients, here is a concise version of what we'll see here:
```

Comment:
Would be good to have a couple of lines explaining what this is about - even for the short version, something like:

Another approach to vendoring consists of invoking the generators programmatically instead of via the `goagen` tool. With this approach a little bit of glue code is used to invoke the code generation functions. The resulting tool is versioned instead of goagen which has the advantage of making the dependencies explicit to the vendoring utility.